src/nvim/api/win_config.c

```diff
@@ -1279,12 +1273,13 @@ static bool parse_win_config(win_T *wp, Dict(win_config) *config, WinConfig *fco
     }
   }
 
-  if (HAS_KEY_X(config, border)) {
+  if (HAS_KEY_X(config, border) || *p_winbd != NUL) {
```

Comment:
This'll prevent creating split windows using `nvim_open_win` whenever `'winborder'` is set.